### ISSUE TITLE
Add filtering by custom section name to wasm-objdump

### ISSUE THREAD

**Issue by Keno** [WebAssembly/wabt#1095]
I have a wasm .so with the following sections:
```
~/wabt/bin/wasm-objdump -h /home/keno/julia-wasm/julia/build-wasm/usr/lib/libjulia.so

libjulia.so:	file format wasm 0x1
023d986: warning: unable to read u32 leb128: index

Sections:

     Type start=0x0000000b end=0x000001cf (size=0x000001c4) count: 63
   Import start=0x000001d3 end=0x0000bc14 (size=0x0000ba41) count: 1692
 Function start=0x0000bc17 end=0x0000c2a5 (size=0x0000068e) count: 1676
     Elem start=0x0000c2a8 end=0x0000c5ed (size=0x00000345) count: 1
     Code start=0x0000c5f1 end=0x000e35b6 (size=0x000d6fc5) count: 1676
     Data start=0x000e35ba end=0x001a5de3 (size=0x000c2829) count: 2187
   Custom start=0x001a5de7 end=0x001bba4c (size=0x00015c65) ".debug_info"
   Custom start=0x001bba4e end=0x001bba5e (size=0x00000010) ".debug_macinfo"
   Custom start=0x001bba61 end=0x001be6f5 (size=0x00002c94) ".debug_loc"
   Custom start=0x001be6f8 end=0x001c1b56 (size=0x0000345e) ".debug_ranges"
   Custom start=0x001c1b59 end=0x001c2184 (size=0x0000062b) ".debug_abbrev"
   Custom start=0x001c2188 end=0x001d6950 (size=0x000147c8) ".debug_line"
   Custom start=0x001d6953 end=0x001d9b72 (size=0x0000321f) ".debug_str"
   Custom start=0x001d9b76 end=0x00201762 (size=0x00027bec) "linking"
   Custom start=0x00201766 end=0x0023d986 (size=0x0003c220) "reloc.CODE"
   Custom start=0x0023d989 end=0x0023e7e8 (size=0x00000e5f) "reloc.DATA"
   Custom start=0x0023e7ec end=0x0024dd7d (size=0x0000f591) "reloc..debug_info"
   Custom start=0x0024dd80 end=0x0024fbbf (size=0x00001e3f) "reloc..debug_loc"
   Custom start=0x0024fbc3 end=0x00253e88 (size=0x000042c5) "reloc..debug_ranges"
   Custom start=0x00253e8b end=0x00254673 (size=0x000007e8) "reloc..debug_line"
   Custom start=0x00254677 end=0x00264554 (size=0x0000fedd) "name"
   Custom start=0x00264557 end=0x002645df (size=0x00000088) "producers"
```

I was expecting this to work:
```
~/wabt/bin/wasm-objdump -x --section "reloc.CODE" /home/keno/julia-wasm/julia/build-wasm/usr/lib/libjulia.so

libjulia.so:	file format wasm 0x1
023d986: warning: unable to read u32 leb128: index

Section Details:

Section not found: reloc.CODE
```

However, the SECTION argument to `--section` only refers to the built in section types. It would be nice if it could filter custom section types by name also.

**Comment by binji** [comment]
Yes, this seems like a good feature. I'll try to add support for this soon, but if you'd like to submit a PR, I'd be happy to review that too.

**Comment by Keno** [comment]
Unfortunately, I'm unlikely to get to it in the near future. This was more of a drive-by feature request.

**Comment by Keno** [comment]
While I'm here, I guess I should also note that the `--section=foo` syntax that appears in the help doesn't seem to actually work (you have to use `--section foo`, so either the parsing or the help should probably be updated.

**Comment by binji** [comment]
Fixed in #1096 and #1097.
